pkg/util/leaderlease/leaderlease.go

```diff
@@ -165,7 +166,10 @@ func (e *Etcd) tryHold(ttl, index uint64) error {
 			return
 		}
 		glog.V(4).Infof("Lease %s lost due to deletion at %d", e.key, watchIndex)
-		close(lost)
+		if !closedLost {
```

Comment:
Am I the only one finding that receiving from a channel before trying to close it is saner than a boolean?